src/coreclr/jit/ssabuilder.cpp

```diff
@@ -563,14 +564,32 @@ void SsaBuilder::InsertPhi(BasicBlock* block, unsigned lclNum)
 void SsaBuilder::AddPhiArg(
     BasicBlock* block, Statement* stmt, GenTreePhi* phi, unsigned lclNum, unsigned ssaNum, BasicBlock* pred)
 {
-#ifdef DEBUG
-    // Make sure it isn't already present: we should only add each definition once.
+    // If there's already a phi arg for this pred, it had better have
+    // matching ssaNum, unless this block is a handler entry.
+    //
+    const bool isHandlerEntry = m_pCompiler->bbIsHandlerBeg(block);
+
     for (GenTreePhi::Use& use : phi->Uses())
     {
-        assert(use.GetNode()->AsPhiArg()->GetSsaNum() != ssaNum);
+        GenTreePhiArg* const phiArg = use.GetNode()->AsPhiArg();
+
+        if (phiArg->gtPredBB == pred)
+        {
+            if (phiArg->GetSsaNum() == ssaNum)
+            {
+                // We already have this (pred, ssaNum) phiArg
+                return;
```

Comment:
Does this situation happen in practice? Seems odd we'd add the same phi arg twice.

Reply:
I was roughly copying what the previous code did. I'll find out if we ever hit this case.

Reply:
Yes, this can happen.

`AddPhiArgsToSuccessors` will enumerate all the successors of a block via `GetAllSuccs`, and for those successors that are try entries, will also add phi args to handlers reachable from the try (since we assume that execution of that try might throw an exception while the def from block is the current def).

But the `GetAllSuccs` enumerator has similar logic for enumerating handlers as successors of a block with a try successor, so we end up redundantly trying to add Phis.

Seems like this is fixable, but best done as its own zero diff change.